### ISSUE TITLE
Coding style and style checks

### ISSUE THREAD

**Issue by adammoody** [LLNL/UnifyCR#212]
This ticket is to discuss code style options.  We can place each suggestion in its own comment block.  Of particular interesting is deciding on a style that is consistent with the existing code base and/or bring the existing code base into compliance.

**Comment by CamStan** [comment]
Can always use the *thumbs up* or *thumbs down* reactions on the top right of each comment to vote for the first or second suggestion.

**Comment by CamStan** [comment]
Suggested style:
```
int foo = function();
if (foo)
```
Checkpatch style:
```
int foo = function();

if (foo)
```

**Comment by CamStan** [comment]
Suggested style:
```
if (thing1 == thing2 &&
    thing3 == thing4)
{
    then do stuff;
}
```
Checkpatch style:
```
if (thing1 == thing2 &&
    thing3 == thing4) {
    then do stuff;
}
```

**Comment by CamStan** [comment]
Suggested style:
```
/* multiline comment
 * describing a code block */
int foo = function();
```
Checkpatch style:
```
/* multiline comment
 * describing a code block
 */
int foo = function();
```

**Comment by CamStan** [comment]
Suggested style:
```
if (foo) {
  code;
}
```
Checkpatch style:
```
if (foo)
  code;
```

**Comment by CamStan** [comment]
Suggested style:
```
if (! foo)
```
Checkpatch style:
```
if (!foo)
```

**Comment by MichaelBrim** [comment]
Note I interpreted "thumbs-up" as voting for suggested style.

**Comment by adammoody** [comment]
Thanks, @CamStan for listing all of these.

I'd say the best option would be to update check patch to check what we decide on. However, that's assuming we can or have the time to figure out the perl foo needed to do that. As a fallback, I'd say it's fine to just carve out particular checks it's currently enforcing that go against what we decide is ok.

I've got a test commit in there against checkpatch.pl to verify that we can do the latter:

baa024b

This hack does the following:
- allows */ to be at end of comment line, rather than requiring it to be on a separate line
- allows use of a variable to follow its declaration without requiring an empty line in between
- allows a space after unary !
- allow braces to exist around if/while/for blocks even for bodies with a single line

**Comment by adammoody** [comment]
Assuming we're ok with that, I've updated https://github.com/LLNL/UnifyCR/pull/199 to fix the rest of the spots where it was failing style checks.

**Comment by MichaelBrim** [comment]
Suggested style:

```
if (CONSTANT == variable)
```

Checkpatch style:

```
if (variable == CONSTANT)
```

Kernel developers seem to like the "natural readability" of the checkpatch style. I prefer the suggested style to avoid accidental assignments (yes, this does happen, people are fallible).

**Comment by CamStan** [comment]
Suggested style:
```
char* foo;
```
Checkpatch style:
```
char *foo;
```
I don't have a particular preference here, but I've noticed we've had PR's fail for at times. Figured I'd throw it in.

**Comment by CamStan** [comment]
@adammoody, I did a little digging and made some changes to checkpatch that should enforce these last two checks that we changed to keep from having one more thing to remember to watch for when reviewing PRs. I have it in a branch on my repo ([here](https://github.com/CamStan/UnifyCR/commit/d4d95d222ec0c54f9458560e27d25d3e546eba8e)) but haven't made a PR yet.

The * position on pointers wasn't too bad and should work fine.

For the order of constant comparisons, the regex for just switching the order starting picking up some false positives so I made it more specific where it now only checks for expressions with just a single comparison (i.e., it will catch `(variable == CONSTANT)` but not `(var == -1 && variable == CONSTANT)`).
This one might be a little more iffy and I can always change it back to allowing either case if we want to play it safe.
